NewBatteryState.h

```diff
@@ -0,0 +1,39 @@
+#include "Battery.h"
+class NewBatteryState : public Battery
+{
+  public:
+    bool checkTempRange(float temperature)
```

Comment:
checkTempRange() is defined as public. Hence asserts maybe added to extend test to test these kind of methods. Overall having test cases to test any public method is a beneficial.

Reply:
I made these function to private